middleware/transaction/unittest/source/test_manager.cpp

```diff
@@ -2012,7 +2012,97 @@ namespace casual
             EXPECT_TRUE( message.trid == trid) << CASUAL_NAMED_VALUE( message);
             EXPECT_TRUE( message.state == decltype( message.state)::ok) << CASUAL_NAMED_VALUE( message.state);
          }
+      }
+
+      namespace local
+      {
+         namespace
+         {
+            void involve_resources( auto& resources, auto& trid)
+            {
+               for( auto& resource : resources)
+               {
+                  common::message::transaction::resource::external::Involved message;
+                  message.process = resource.process;
+                  message.trid = trid;
+                  local::send::tm( message);
+               }
+            };
+
+            void handle_rollback_request( auto& resources, auto& trid)
+            {
+               for( auto& resource : resources)
+               {
+                  auto request = common::communication::device::receive< common::message::transaction::resource::rollback::Request>( resource.inbound);
+                  auto reply = common::message::reverse::type( request);
+                  reply.trid = request.trid;
+                  reply.resource = request.resource;
+                  reply.state = common::code::xa::ok;
+                  local::send::tm( reply);
+               }
+            };
+
+            
+         } // <unnamed>
+      } // local
+
+      TEST( transaction_manager, two_resources_involved__send_potential_stale__expect_rollback)
+      {
+         common::unittest::Trace trace;
+
+         auto domain = local::domain( local::configuration::system, local::configuration::base);
+
+         auto resources = std::array< local::involved::Process, 2>();
+
+         auto trid = common::transaction::id::create( process::handle());
+
+         local::involve_resources( resources, trid);
+
+
+         // send the potentially stale message, this will trigger a rollback.
+         {
+            common::message::transaction::potential::Stale message{ process::handle()};
+            message.gtrid = common::transaction::id::range::global( trid);
+            local::send::tm( message);
+         }
+
+         local::handle_rollback_request( resources, trid);
```

Comment:
How would this test fail? By hanging indefinitely?

Reply:
Yes.

Reply:
Do you think we need to no-block pull just in case?

Reply:
I don't know about need, but I'd argue it's somewhat more clear what went wrong if we get a "normal" test failure.